server/routes/data_connections/data_connections_router.ts

```diff
@@ -103,6 +103,43 @@ export function registerDataConnectionsRoute(router: IRouter) {
     }
   );
 
+  router.post(
+    {
+      path: `${DATACONNECTIONS_BASE}`,
+      validate: {
+        body: schema.object({
+          name: schema.string(),
+          connector: schema.string(),
+          allowedRoles: schema.arrayOf(schema.string()),
+          properties: schema.any(),
+        }),
+      },
+    },
+    async (context, request, response): Promise<any> => {
```

Comment:
probably just remove the type if it's any?
```suggestion
    async (context, request, response) => {
```

Reply:
Addressed in: [680763b](https://github.com/opensearch-project/dashboards-observability/pull/1049/commits/680763b566a24a70d93f79214331182f495ed291) - I think later on I need to lock down the types in a similar way to the datasources - but need to figure out what is the structure of the datasources properties for the different types we will support.